### PR TITLE
[GEOS-10631] AccessManager will not be looked up if multiple beans are of type DefaultResourceAccessManager

### PR DESCRIPTION
**Pull request by taba90**
[GEOS-10631](https://osgeo-org.atlassian.net/browse/GEOS-10631)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

File: src/main/src/main/java/org/geoserver/security/SecureCatalogImpl.java [hunk 1]
```diff
@@ -104,16 +104,20 @@ static ResourceAccessManager lookupResourceAccessManager() throws Exception {
...
-                if (!(resourceAccessManager instanceof DefaultResourceAccessManager)) {
+                if (!DefaultResourceAccessManager.class.equals(resourceAccessManager.getClass())) {
                     manager = resourceAccessManager;
                     break;
                 }
             }
         }
+        // should never happen,just in case we have multiple singleton beans
+        // of type DefaultResourceAccessManager
+        if (manager == null) manager = managers.get(size - 1);
```
Comment (aaime) [inline src/main/src/main/java/org/geoserver/security/SecureCatalogImpl.java:120]: Maybe better to use managers.get(0), if multiple managers are registered the first one should be the one with the highest priority (if the ExtensionPriority interface has been used)

**Review by aaime — CHANGES_REQUESTED** [review]
Looks good, just a note about the last resort lookup below

File: src/main/src/main/java/org/geoserver/security/SecureCatalogImpl.java (hunk 1 above)
Comment (taba90) [reply src/main/src/main/java/org/geoserver/security/SecureCatalogImpl.java]: fixed

**Comment by taba90** [comment]
@aaime  thanks for the review, feedback applied

**Pull request merged**